CRM/Contribute/BAO/Contribution.php

```diff
@@ -3592,6 +3604,28 @@ public static function updateFinancialAccounts(&$params, $context = NULL, $skipT
         }
       }
     }
+    if ($context == 'changePaymentInstrument') {
```

Comment:
Is this as per the dataflow for accounting? 
https://wiki.civicrm.org/confluence/display/CRM/CiviAccounts+Data+Flow#CiviAccountsDataFlow-ChangePaymentInstrumentforaPartlyPaidorCompletedcontribution